### ISSUE TITLE
WP_PJAX_PageCache::clearCache implemenation is bugos

### ISSUE THREAD

**Issue by eimanavicius** [pelmered/wp-pjax#8]
https://github.com/pelmered/wp-pjax/blob/b9f4abfc38a768b7b17fe392d89fd59c2efb99e3/inc/PageCache.php#L159

phpcs: WordPress.VIP.DirectDatabaseQuery: Usage of a direct database call is discouraged.

Deleting "_transient_%" options from database does not mean cache cleared. Other back-end can be used (ex. memcache, redis).
Cache versions can be used to invalidate cache. Version would be stored in cache and changed when clearing cache.

**Comment by pelmered** [comment]
Good point. The only way to do this must be to make a select query to get all the transient keys and then loop through them and then run `delete_transient()` on each one. Not very pretty, but that would allow any backend to handle this. 

Or do you have any suggestions/ideas?

The page cache functionality nerver really worked very good, and should probably get a rewrite as well.

**Comment by eimanavicius** [comment]
Well selecting all key would not help as database is not the only possible cache back-end.

My suggestion would be:
> Cache version can be used to invalidate cache. Version would be stored in cache and changed when clearing cache.

This technique is used in Doctrine 2 ORM caching solution.

**Comment by pelmered** [comment]
You mean something like generating keys like `prefix_<version (generated random string)>_<key>` and then generate a new version string when you clear the cache? For example: `wp_pjax_pc-876b438o-test-page`
That would be a good solution.

**Comment by eimanavicius** [comment]
Yes, exactly
